.travis.yml

```diff
@@ -10,32 +13,53 @@ before_install:
   - cat /proc/cpuinfo | grep cores | wc -l
   - free -h
 install:
-  - hibench=$(pwd)
-  - cd /opt/
-  - wget http://d3kbcqa49mib13.cloudfront.net/spark-1.6.0-bin-hadoop2.6.tgz
-  - tar -xzf spark-1.6.0-bin-hadoop2.6.tgz
-  - wget https://archive.apache.org/dist/hadoop/core/hadoop-2.6.5/hadoop-2.6.5.tar.gz
-  - tar -xzf hadoop-2.6.5.tar.gz
-  - cd ${hibench}
-  - cp ./travis/spark-env.sh /opt/spark-1.6.0-bin-hadoop2.6/conf/
-  - cp ./travis/core-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/hdfs-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/mapred-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/yarn-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/hibench.conf ./conf/
-  - cp ./travis/benchmarks.lst ./conf/
+  - |
+    export java_ver=$(./travis/jdk_ver.sh)
+    if [[ "$java_ver" == 11 ]]; then
+        export HADOOP_VER=3.2.0
+        export SPARK_VER=2.4.3
+        export SPARK_PACKAGE_TYPE=without-hadoop-scala-2.12
+    elif [[ "$java_ver" == 8 ]]; then
+        export HADOOP_VER=3.2.0
+        export SPARK_VER=2.4.3
+        export SPARK_PACKAGE_TYPE=without-hadoop
+    elif [[ "$java_ver" == 7 ]]; then
+        export HADOOP_VER=2.6.5
+        export SPARK_VER=1.6.0
+        export SPARK_PACKAGE_TYPE=hadoop2.6
+    else
+        exit 1
+    fi
+
+    # Folders where are stored Spark and Hadoop depending on version required
+    export SPARK_BINARIES_FOLDER=spark-$SPARK_VER-bin-$SPARK_PACKAGE_TYPE
+    export HADOOP_BINARIES_FOLDER=hadoop-$HADOOP_VER
+    export HADOOP_CONF_DIR=/opt/$HADOOP_BINARIES_FOLDER/etc/hadoop/
+    export HADOOP_HOME=/opt/$HADOOP_BINARIES_FOLDER
+
+    sudo -E ./travis/install_hadoop_spark.sh
+    sudo -E ./travis/config_hadoop_spark.sh
 before_script:
   - "export JAVA_OPTS=-Xmx512m"
 cache:
   directories:
   - $HOME/.m2
 script:
-  - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.2 -Dscala=2.11
-  - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.0 -Dscala=2.11
-  - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=1.6 -Dscala=2.10
-  - sudo -E ./travis/configssh.sh
-  - sudo -E ./travis/restart_hadoop_spark.sh
-  - cp ./travis/hadoop.conf ./conf/
-  - cp ./travis/spark.conf ./conf/
-  - /opt/hadoop-2.6.5/bin/yarn node -list 2
-  - sudo -E ./bin/run_all.sh
+  - |
+    if [[ "$java_ver" == 11 ]]; then
+        mvn clean package -Psparkbench -Phadoopbench -Dhadoop=3.2 -Dspark=2.4 -Dscala=2.12 -Dmaven-compiler-plugin.version=3.8.0 -Dexclude-streaming
+    elif [[ "$java_ver" == 8 ]]; then
+        mvn clean package -q -Dmaven.javadoc.skip=true -Dhadoop=3.2 -Dspark=2.4 -Dscala=2.11
```

Comment:
Curious about this, even if we don't run SQL module tests for Spark2.4, how did the compiling work...